### PR TITLE
User Auth for Client and Server

### LINKED ISSUE #17: Server and Client User Auth setup
Server:
- Set up user authentication tokens.
- Create login endpoint for users.

Client:
- Add user store with routes configured
- Add beta login
- Add browser token storage

We should make user login secured with JWTs and redirect users who are not logged in to the login page.

### PR DESCRIPTION
**Pull request by Linkaloo**
Closes #17 

Use JWT tokens for user authentication.
Server signs a token for the client to use.
Upon certain requests, the server verifies against the token.
Added prettier to server and formatted with it.

Client redirects user to login page from home screen if not logged in.
Client retrieves a token on login and stores it in a cookie.
Client redirects user from login page to home page on successful login.

### CODE REVIEW

**Review by dafneMena — APPROVED** [review]
This looks good. I read your message re the minor login page. Once this gets approved and Aligary finishes the button, I'll go ahead and add the other small details to the page. This is more of a note to self, but the login should include: 

- username text field (or email) 
- password text field 
- login button
- updated svg logo 
- 'forgot password' text 
- 'register account' text

**Review by tbmcallister — APPROVED** [review]
Code is easy to follow, It could use some more comments, but I'm not going to block the review because of that(simple fix for another time), It doesnt seem like your code is doing anything other than its intended purpose, and the auth API logic is sound.

File: client/src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -1,9 +1,46 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import { useState } from 'react';
 import styles from './Login.module.scss';
+import { TextField, Button } from '@mui/material';
```
Comment (Aligary) [inline client/src/pages/Login/Login.tsx:4]: use the custom components made for TextField

**Review by Aligary — APPROVED** [review]
Everything looks good. Just had 1 comment.

**Pull request merged**